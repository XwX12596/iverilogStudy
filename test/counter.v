module counter
#(parameter N = 4)
(
    input wire clk, rst,
    output wire [N-1:0] bin
);

reg [N-1:0]r;
wire [N-1:0]r_next;

always @(posedge clk or negedge rst) begin
    if(~rst)
        r <= 0;
    else
        r <= r_next;
end
assign bin = r;
assign r_next = r + 1;
endmodule