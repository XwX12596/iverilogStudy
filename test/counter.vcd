$date
	Tue Sep 20 18:07:52 2022
$end
$version
	Icarus Verilog
$end
$timescale
	1ps
$end
$scope module stimulus $end
$var wire 4 ! binary [3:0] $end
$var reg 1 " clk $end
$var reg 1 # reset $end
$var integer 32 $ i [31:0] $end
$scope module u_bc $end
$var wire 1 % Cout $end
$var wire 4 & binary [3:0] $end
$var wire 1 " clk $end
$var wire 1 # reset $end
$var wire 4 ' r_next [3:0] $end
$var parameter 32 ( N $end
$var reg 1 ) C $end
$var reg 4 * r_reg [3:0] $end
$upscope $end
$upscope $end
$enddefinitions $end
$comment Show the parameter values. $end
$dumpall
b100 (
$end
#0
$dumpvars
bx *
0)
bx '
bx &
0%
b0 $
1#
0"
bx !
$end
#2000
b1 '
b0 !
b0 &
b0 *
0#
#4000
1#
#10000
b10 '
b1 !
b1 &
b1 *
b1 $
1"
#20000
b10 $
0"
#30000
b11 '
b10 !
b10 &
b10 *
b11 $
1"
#40000
b100 $
0"
#50000
b100 '
b11 !
b11 &
b11 *
b101 $
1"
#60000
b110 $
0"
#70000
b101 '
b100 !
b100 &
b100 *
b111 $
1"
#80000
b1000 $
0"
#90000
b110 '
b101 !
b101 &
b101 *
b1001 $
1"
#100000
b1010 $
0"
#110000
b111 '
b110 !
b110 &
b110 *
b1011 $
1"
#120000
b1100 $
0"
#130000
b1000 '
b111 !
b111 &
b111 *
b1101 $
1"
#140000
b1110 $
0"
#150000
b1001 '
b1000 !
b1000 &
b1000 *
b1111 $
1"
#160000
b10000 $
0"
#170000
b1010 '
b1001 !
b1001 &
b1001 *
b10001 $
1"
#180000
b10010 $
0"
#190000
b1011 '
b1010 !
b1010 &
b1010 *
b10011 $
1"
#200000
b10100 $
0"
#210000
b1100 '
b1011 !
b1011 &
b1011 *
b10101 $
1"
#220000
b10110 $
0"
#230000
b1101 '
b1100 !
b1100 &
b1100 *
b10111 $
1"
#240000
b11000 $
0"
#250000
b1110 '
b1101 !
b1101 &
b1101 *
b11001 $
1"
#260000
b11010 $
0"
#270000
b1111 '
b1110 !
b1110 &
b1110 *
b11011 $
1"
#280000
b11100 $
0"
#290000
1%
1)
b0 '
b1111 !
b1111 &
b1111 *
b11101 $
1"
#300000
b11110 $
0"
#310000
0%
0)
b1 '
b0 !
b0 &
b0 *
b11111 $
1"
#320000
b100000 $
0"
#330000
b10 '
b1 !
b1 &
b1 *
b100001 $
1"
#340000
b100010 $
0"
#350000
b11 '
b10 !
b10 &
b10 *
b100011 $
1"
#360000
b100100 $
0"
#370000
b100 '
b11 !
b11 &
b11 *
b100101 $
1"
#380000
b100110 $
0"
#390000
b101 '
b100 !
b100 &
b100 *
b100111 $
1"
#400000
b101000 $
0"
#410000
b110 '
b101 !
b101 &
b101 *
b101001 $
1"
