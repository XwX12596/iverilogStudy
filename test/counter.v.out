#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 12;
:vpi_module "D:\iverilog\lib\ivl\system.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "D:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "D:\iverilog\lib\ivl\va_math.vpi";
S_00000148492bbd60 .scope module, "stimulus" "stimulus" 2 32;
 .timescale -9 -12;
v000001484934d8c0_0 .net "binary", 3 0, L_00000148492bdf30;  1 drivers
v000001484934d960_0 .var "clk", 0 0;
v000001484934da00_0 .var/i "i", 31 0;
v000001484934daa0_0 .var "reset", 0 0;
S_00000148492bdb20 .scope module, "u_bc" "bc" 2 36, 2 1 0, S_00000148492bbd60;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "clk";
    .port_info 1 /INPUT 1 "reset";
    .port_info 2 /OUTPUT 4 "binary";
    .port_info 3 /OUTPUT 1 "Cout";
P_0000014849308f50 .param/l "N" 0 2 2, +C4<00000000000000000000000000000100>;
L_00000148492bdf30 .functor BUFZ 4, v00000148492bde90_0, C4<0000>, C4<0000>, C4<0000>;
L_0000014849395cb0 .functor BUFZ 1, v00000148492b6df0_0, C4<0>, C4<0>, C4<0>;
v00000148492b6df0_0 .var "C", 0 0;
v00000148492ba8f0_0 .net "Cout", 0 0, L_0000014849395cb0;  1 drivers
L_000001484934db48 .functor BUFT 1, C4<0001>, C4<0>, C4<0>, C4<0>;
v00000148492ba610_0 .net/2u *"_ivl_2", 3 0, L_000001484934db48;  1 drivers
v00000148492bdcb0_0 .net "binary", 3 0, L_00000148492bdf30;  alias, 1 drivers
v00000148492bdd50_0 .net "clk", 0 0, v000001484934d960_0;  1 drivers
v00000148492bddf0_0 .net "r_next", 3 0, L_0000014849395b50;  1 drivers
v00000148492bde90_0 .var "r_reg", 3 0;
v000001484934d820_0 .net "reset", 0 0, v000001484934daa0_0;  1 drivers
E_0000014849309190/0 .event negedge, v000001484934d820_0;
E_0000014849309190/1 .event posedge, v00000148492bdd50_0;
E_0000014849309190 .event/or E_0000014849309190/0, E_0000014849309190/1;
L_0000014849395b50 .arith/sum 4, v00000148492bde90_0, L_000001484934db48;
    .scope S_00000148492bdb20;
T_0 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00000148492b6df0_0, 0, 1;
    %end;
    .thread T_0;
    .scope S_00000148492bdb20;
T_1 ;
    %wait E_0000014849309190;
    %load/vec4 v000001484934d820_0;
    %inv;
    %flag_set/vec4 8;
    %jmp/0xz  T_1.0, 8;
    %pushi/vec4 0, 0, 4;
    %assign/vec4 v00000148492bde90_0, 0;
    %pushi/vec4 0, 0, 1;
    %assign/vec4 v00000148492b6df0_0, 0;
    %jmp T_1.1;
T_1.0 ;
    %load/vec4 v00000148492bddf0_0;
    %assign/vec4 v00000148492bde90_0, 0;
    %load/vec4 v00000148492bddf0_0;
    %cmpi/e 15, 0, 4;
    %jmp/0xz  T_1.2, 4;
    %pushi/vec4 1, 0, 1;
    %assign/vec4 v00000148492b6df0_0, 0;
    %jmp T_1.3;
T_1.2 ;
    %pushi/vec4 0, 0, 1;
    %assign/vec4 v00000148492b6df0_0, 0;
T_1.3 ;
T_1.1 ;
    %jmp T_1;
    .thread T_1;
    .scope S_00000148492bbd60;
T_2 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v000001484934d960_0, 0, 1;
    %pushi/vec4 0, 0, 32;
    %store/vec4 v000001484934da00_0, 0, 32;
T_2.0 ;
    %load/vec4 v000001484934da00_0;
    %cmpi/s 40, 0, 32;
    %flag_or 5, 4;
    %jmp/0xz T_2.1, 5;
    %delay 10000, 0;
    %load/vec4 v000001484934d960_0;
    %inv;
    %store/vec4 v000001484934d960_0, 0, 1;
    %load/vec4 v000001484934da00_0;
    %addi 1, 0, 32;
    %store/vec4 v000001484934da00_0, 0, 32;
    %jmp T_2.0;
T_2.1 ;
    %end;
    .thread T_2;
    .scope S_00000148492bbd60;
T_3 ;
    %vpi_call 2 51 "$dumpfile", "counter.vcd" {0 0 0};
    %vpi_call 2 52 "$dumpvars", 32'sb00000000000000000000000000000000, S_00000148492bbd60 {0 0 0};
    %pushi/vec4 1, 0, 1;
    %store/vec4 v000001484934daa0_0, 0, 1;
    %delay 2000, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v000001484934daa0_0, 0, 1;
    %delay 2000, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v000001484934daa0_0, 0, 1;
    %end;
    .thread T_3;
# The file index is used to find the file name in the following table.
:file_names 3;
    "N/A";
    "<interactive>";
    "counter.v";
